plugins/net_plugin/net_plugin.cpp

```diff
@@ -3190,8 +3191,10 @@ namespace eosio {
             return true;
       }
 
-      auto ds = pending_message_buffer.create_datastream();
-      fc::raw::unpack( ds, which );
+      auto mb_ds = pending_message_buffer.create_datastream();
+      fc::raw::unpack( mb_ds, which );
+
+      fc::datastream_mirror ds(mb_ds, message_length);
```

Comment:
cool, good dimensioning of the vector here as well. I see that my suggestion on the `resize()` was not as critical  as I originally thought.